primitives/traits/src/lib.rs

```diff
@@ -38,9 +38,16 @@ pub trait ProvidersInterface {
         + Member
         + MaybeSerializeDeserialize
         + Debug
+        + MaybeDisplay
+        + SimpleBitOps
         + Ord
+        + Default
+        + Copy
+        + CheckEqual
+        + AsRef<[u8]>
+        + AsMut<[u8]>
         + MaxEncodedLen
-        + Copy;
+        + FullCodec;
```

Comment:
What is all these needed for?

Reply:
This is because I now ensure that `BucketId` and `ProviderId` are the same type (hashes). This requirement was added because now the file system pallet verifies the forest proof. And the forest verifier only accepts `ProviderId`s as an identifier. Working with bucket's and providers should all share the same type under these conventions.

Reply:
To answer your question specifically, the trait bounds were not enough for rust to figure out that a bucket id (being a provider id) has the `as_ref()` method. So I copied exactly the same traits from `BucketId` to `ProviderId`.